Running the app via npm breaks auto-rotation

Not sure if its a problem with just my machine. But If I startup the app using `npm run start`, or `npm run start:dev` I get the following runtime error when auto rotation happens.

```
Error Launching App:

Unable to find Electron app at /Users/jameswalsh/dev/channels/rotate

Cannot find module '/Users/jameswalsh/dev/channels/rotate'
```

* Do I need to reinitialize my repos since auto-rotation changes have been made?
* I should also add this does not happen when I package up the app and run it from the desktop instead of the command line.

--- a/README.md
+++ b/README.md
@@ -11,7 +11,7 @@ Cross-platform electron app for managing git users while pair/mob programming
 ## How it works
 Git Switch adds a post commit hook to the git repositories you specify.
 
-You select the users to add to you pair/mob, and commit changes to your code.
+You select the users to add to your pair/mob, and commit changes to your code.
 
 With each commit, the git-switch commit hook amends the commit to designate the author separate from the committer(s).
 

--- a/scripts/utils/package.js
+++ b/scripts/utils/package.js
@@ -13,11 +13,11 @@ function removePackageSrc(os) {
 }
 
 function getPackageSrcDir(os) {
-  return path.join(__dirname, '../../out', packageRelease[os])
+  return path.join(__dirname, '..', '..', 'out', packageRelease[os])
 }
 
 function getPackageZipDir(os) {
-  return path.join(__dirname, '../../out/', `git-switch-${os}.zip`)
+  return path.join(__dirname, '..', '..', 'out', `git-switch-${os}.zip`)
 }
 
 module.exports = {

--- a/src/common/utils/__specs__/install.spec.js
+++ b/src/common/utils/__specs__/install.spec.js
@@ -34,7 +34,6 @@ describe('utils/install', () => {
   let appExecutablePath
   let autoRotate
   let platform
-  let isDev
   let postCommitFileContents
 
   beforeEach(() => {
@@ -44,7 +43,6 @@ describe('utils/install', () => {
     appExecutablePath = '/foo/bar'
     autoRotate = '/foo/bar rotate > /dev/null 2>&1 &'
     platform = 'linux'
-    isDev = false
     postCommitFileContents = getPostCommitFileContents(autoRotate)
 
     sinon.stub(fs, 'existsSync')
@@ -65,7 +63,7 @@ describe('utils/install', () => {
       gitSwitchDirExists = false
       sinon.stub(fs, 'mkdirSync')
 
-      subject(platform, appExecutablePath, isDev)
+      subject(platform, appExecutablePath)
 
       expect(fs.mkdirSync).to.have.been.calledWith(GIT_SWITCH_PATH)
     })
@@ -80,7 +78,7 @@ describe('utils/install', () => {
       configFileExsists = false
       sinon.stub(fs, 'writeFileSync')
 
-      subject(platform, appExecutablePath, isDev)
+      subject(platform, appExecutablePath)
 
       expect(fs.writeFileSync).to.have.been.calledWith(CONFIG_FILE, JSON.stringify({ users: [], repos: [] }), 'utf-8')
     })
@@ -98,18 +96,21 @@ describe('utils/install', () => {
     })
 
     it('creates .git-switch/post-commit', () => {
-      subject(platform, appExecutablePath, isDev)
+      subject(platform, appExecutablePath)
       expect(fs.writeFileSync).to.have.been.calledWith(POST_COMMIT_FILE, postCommitFileContents, 'utf-8')
     })
 
-    describe('when isDev is true', () => {
+    describe('when app executable is electron', () => {
       beforeEach(() => {
-        isDev = true
-        postCommitFileContents = getPostCommitFileContents('echo "git-switch > Auto-rotate is disabled when running from npm"')
+        appExecutablePath = '/herp/derp/node_modules/electron-prebuilt-compile/node_modules/dist/electron'
+        autoRotate = `cd /herp/derp
+  npm run start --- -- rotate
+  cd $(dirname $0)/../../`
+        postCommitFileContents = getPostCommitFileContents(autoRotate)
       })
 
-      it('the post-commit file echoes a message rather than auto-rotating', () => {
-        subject(platform, appExecutablePath, isDev)
+      it('the post-commit file auto rotates by changing dirs and running npm', () => {
+        subject(platform, appExecutablePath)
         expect(fs.writeFileSync).to.have.been.calledWith(POST_COMMIT_FILE, postCommitFileContents, 'utf-8')
       })
     })
@@ -123,7 +124,7 @@ describe('utils/install', () => {
       })
 
       it('escapes and backgrounds the autoRotate specific to the platform', () => {
-        subject(platform, appExecutablePath, isDev)
+        subject(platform, appExecutablePath)
         expect(fs.writeFileSync).to.have.been.calledWith(POST_COMMIT_FILE, postCommitFileContents, 'utf-8')
       })
     })
@@ -147,12 +148,12 @@ describe('utils/install', () => {
     })
 
     it('updates .git-switch/post-commit', () => {
-      subject(platform, appExecutablePath, isDev)
+      subject(platform, appExecutablePath)
       expect(fs.writeFileSync).to.have.been.calledWith(POST_COMMIT_FILE, postCommitFileContents, 'utf-8')
     })
 
     it('re-initializes all the repos', () => {
-      subject(platform, appExecutablePath, isDev)
+      subject(platform, appExecutablePath)
 
       expect(repoService.add).to.have.been.calledWith('repo/one')
       expect(repoService.add).to.have.been.calledWith('repo/two')

--- a/src/common/utils/__specs__/install.spec.js
+++ b/src/common/utils/__specs__/install.spec.js
@@ -6,22 +6,8 @@ import * as repoService from '../../services/repo'
 
 import subject, { GIT_SWITCH_PATH, CONFIG_FILE, POST_COMMIT_FILE } from '../install'
 
-describe('utils/install', () => {
-  let gitSwitchDirExists
-  let configFileExsists
-  let postCommitFileExists
-  let appExecutablePath
-  let isDev
-  let postCommitFileContents
-  let existingPostCommitFileContents
-
-  beforeEach(() => {
-    gitSwitchDirExists = true
-    configFileExsists = true
-    postCommitFileExists = true
-    appExecutablePath = 'C:\\foo\\bar'
-    isDev = false
-    postCommitFileContents = `#!/bin/sh
+function getPostCommitFileContents(autoRotate) {
+  return `#!/bin/sh
 
 actual_author=$(git log -1 HEAD --format="%an")
 expected_author=$(git config --global author.name)
@@ -34,21 +20,40 @@ if [ "$actual_author" != "$expected_author" ]; then
   echo ""
 
   git commit --amend --no-verify --no-edit --author="$expected_author <$expected_author_email>"
-  C:\\\\foo\\\\bar rotate
+  echo ""
+  echo "git-switch > Rotating author and committer(s)"
+  ${autoRotate}
 fi
 `
-    existingPostCommitFileContents = postCommitFileContents
+}
+
+describe('utils/install', () => {
+  let gitSwitchDirExists
+  let configFileExsists
+  let postCommitFileExists
+  let appExecutablePath
+  let autoRotate
+  let platform
+  let isDev
+  let postCommitFileContents
+
+  beforeEach(() => {
+    gitSwitchDirExists = true
+    configFileExsists = true
+    postCommitFileExists = true
+    appExecutablePath = '/foo/bar'
+    autoRotate = '/foo/bar rotate > /dev/null 2>&1 &'
+    platform = 'linux'
+    isDev = false
+    postCommitFileContents = getPostCommitFileContents(autoRotate)
 
     sinon.stub(fs, 'existsSync')
       .withArgs(GIT_SWITCH_PATH).callsFake(() => gitSwitchDirExists)
       .withArgs(CONFIG_FILE).callsFake(() => configFileExsists)
       .withArgs(POST_COMMIT_FILE).callsFake(() => postCommitFileExists)
-
-    sinon.stub(fs, 'readFileSync').callsFake(() => existingPostCommitFileContents)
   })
   afterEach(() => {
     fs.existsSync.restore()
-    fs.readFileSync.restore()
   })
 
   describe('when config directory does not exist', () => {
@@ -60,7 +65,7 @@ fi
       gitSwitchDirExists = false
       sinon.stub(fs, 'mkdirSync')
 
-      subject(appExecutablePath, isDev)
+      subject(platform, appExecutablePath, isDev)
 
       expect(fs.mkdirSync).to.have.been.calledWith(GIT_SWITCH_PATH)
     })
@@ -75,7 +80,7 @@ fi
       configFileExsists = false
       sinon.stub(fs, 'writeFileSync')
 
-      subject(appExecutablePath, isDev)
+      subject(platform, appExecutablePath, isDev)
 
       expect(fs.writeFileSync).to.have.been.calledWith(CONFIG_FILE, JSON.stringify({ users: [], repos: [] }), 'utf-8')
     })
@@ -93,59 +98,61 @@ fi
     })
 
     it('creates .git-switch/post-commit', () => {
-      subject(appExecutablePath, isDev)
+      subject(platform, appExecutablePath, isDev)
       expect(fs.writeFileSync).to.have.been.calledWith(POST_COMMIT_FILE, postCommitFileContents, 'utf-8')
     })
 
     describe('when isDev is true', () => {
       beforeEach(() => {
-        postCommitFileExists = false
         isDev = true
-        postCommitFileContents = postCommitFileContents = `#!/bin/sh
-
-actual_author=$(git log -1 HEAD --format="%an")
-expected_author=$(git config --global author.name)
-expected_author_email=$(git config --global author.email)
-committers=$(git config --global user.name)
-
-if [ "$actual_author" != "$expected_author" ]; then
-  echo "git-switch > Author: $expected_author"
-  echo "git-switch > Committer(s): $committers"
-  echo ""
-
-  git commit --amend --no-verify --no-edit --author="$expected_author <$expected_author_email>"
-  echo "git-switch > Auto-rotate is disabled when running from npm"
-fi
-`
+        postCommitFileContents = getPostCommitFileContents('echo "git-switch > Auto-rotate is disabled when running from npm"')
       })
 
       it('the post-commit file echoes a message rather than auto-rotating', () => {
-        subject(appExecutablePath, isDev)
+        subject(platform, appExecutablePath, isDev)
+        expect(fs.writeFileSync).to.have.been.calledWith(POST_COMMIT_FILE, postCommitFileContents, 'utf-8')
+      })
+    })
+
+    describe('when platform is windows', () => {
+      beforeEach(() => {
+        platform = 'win32'
+        appExecutablePath = 'C:\\foo\\bar'
+        autoRotate = 'start C:\\\\foo\\\\bar rotate'
+        postCommitFileContents = getPostCommitFileContents(autoRotate)
+      })
+
+      it('escapes and backgrounds the autoRotate specific to the platform', () => {
+        subject(platform, appExecutablePath, isDev)
         expect(fs.writeFileSync).to.have.been.calledWith(POST_COMMIT_FILE, postCommitFileContents, 'utf-8')
       })
     })
   })
 
   describe('when post-commit file is outdated', () => {
+    let existingPostCommitFileContents
+
     beforeEach(() => {
       existingPostCommitFileContents = 'outdated-content-here'
       sinon.stub(repoService, 'add')
       sinon.stub(repoService, 'get').returns([{ path: 'repo/one' }, { path: 'repo/two' }])
       sinon.stub(fs, 'writeFileSync')
+      sinon.stub(fs, 'readFileSync').callsFake(() => existingPostCommitFileContents)
     })
     afterEach(() => {
       repoService.add.restore()
       repoService.get.restore()
       fs.writeFileSync.restore()
+      fs.readFileSync.restore()
     })
 
     it('updates .git-switch/post-commit', () => {
-      subject(appExecutablePath, isDev)
+      subject(platform, appExecutablePath, isDev)
       expect(fs.writeFileSync).to.have.been.calledWith(POST_COMMIT_FILE, postCommitFileContents, 'utf-8')
     })
 
     it('re-initializes all the repos', () => {
-      subject(appExecutablePath, isDev)
+      subject(platform, appExecutablePath, isDev)
 
       expect(repoService.add).to.have.been.calledWith('repo/one')
       expect(repoService.add).to.have.been.calledWith('repo/two')

--- a/src/common/utils/install.js
+++ b/src/common/utils/install.js
@@ -8,9 +8,11 @@ export const GIT_SWITCH_PATH = path.join(os.homedir(), '.git-switch')
 export const CONFIG_FILE = path.join(GIT_SWITCH_PATH, 'config.json')
 export const POST_COMMIT_FILE = path.join(GIT_SWITCH_PATH, 'post-commit')
 
-export default function(appExecutablePath, isDev) {
+export default function(platform, appExecutablePath, isDev) {
   installConfigFile()
-  installPostCommitHook(appExecutablePath, isDev)
+
+  const autoRotate = getAutoRotateCommand(platform, appExecutablePath, isDev)
+  installPostCommitHook(autoRotate)
 }
 
 function installConfigFile() {
@@ -23,14 +25,22 @@ function installConfigFile() {
   }
 }
 
-function getAutoRotateCommand(appExecutablePath, isDev) {
-  return isDev
-    ? 'echo "git-switch > Auto-rotate is disabled when running from npm"'
-    : `${appExecutablePath.replace(new RegExp(/\\/, 'g'), '\\\\')} rotate`
+function getAutoRotateCommand(platform, appExecutablePath, isDev) {
+  if (isDev) {
+    return 'echo "git-switch > Auto-rotate is disabled when running from npm"'
+  }
+
+  let prepend = ''
+  let postpend = ' > /dev/null 2>&1 &'
+  if (platform === 'win32') {
+    prepend = 'start '
+    postpend = ''
+  }
+
+  return `${prepend}${appExecutablePath.replace(new RegExp(/\\/, 'g'), '\\\\')} rotate${postpend}`
 }
 
-function installPostCommitHook(appExecutablePath, isDev) {
-  const autoRotate = getAutoRotateCommand(appExecutablePath, isDev)
+function installPostCommitHook(autoRotate) {
   const postCommitScript = `#!/bin/sh
 
 actual_author=$(git log -1 HEAD --format="%an")
@@ -44,6 +54,8 @@ if [ "$actual_author" != "$expected_author" ]; then
   echo ""
 
   git commit --amend --no-verify --no-edit --author="$expected_author <$expected_author_email>"
+  echo ""
+  echo "git-switch > Rotating author and committer(s)"
   ${autoRotate}
 fi
 `

--- a/src/common/utils/install.js
+++ b/src/common/utils/install.js
@@ -8,10 +8,10 @@ export const GIT_SWITCH_PATH = path.join(os.homedir(), '.git-switch')
 export const CONFIG_FILE = path.join(GIT_SWITCH_PATH, 'config.json')
 export const POST_COMMIT_FILE = path.join(GIT_SWITCH_PATH, 'post-commit')
 
-export default function(platform, appExecutablePath, isDev) {
+export default function(platform, appExecutablePath) {
   installConfigFile()
 
-  const autoRotate = getAutoRotateCommand(platform, appExecutablePath, isDev)
+  const autoRotate = getAutoRotateCommand(platform, appExecutablePath)
   installPostCommitHook(autoRotate)
 }
 
@@ -25,9 +25,11 @@ function installConfigFile() {
   }
 }
 
-function getAutoRotateCommand(platform, appExecutablePath, isDev) {
-  if (isDev) {
-    return 'echo "git-switch > Auto-rotate is disabled when running from npm"'
+function getAutoRotateCommand(platform, appExecutablePath) {
+  if (path.basename(appExecutablePath) === 'electron') {
+    return `cd ${appExecutablePath.split('/node_modules')[0]}
+  npm run start --- -- rotate
+  cd $(dirname $0)/../../`
   }
 
   let prepend = ''

--- a/src/main.js
+++ b/src/main.js
@@ -22,7 +22,7 @@ const mb = menubar({
 })
 
 const appExecutablePath = mb.app.getPath('exe')
-install(appExecutablePath, isDev)
+install(process.platform, appExecutablePath, isDev)
 
 const isSecondInstance = mb.app.makeSingleInstance(processAppArgs)
 if (isSecondInstance) {

--- a/src/main.js
+++ b/src/main.js
@@ -22,7 +22,7 @@ const mb = menubar({
 })
 
 const appExecutablePath = mb.app.getPath('exe')
-install(process.platform, appExecutablePath, isDev)
+install(process.platform, appExecutablePath)
 
 const isSecondInstance = mb.app.makeSingleInstance(processAppArgs)
 if (isSecondInstance) {
@@ -54,8 +54,8 @@ function handleAfterCreateWindow() {
 function processAppArgs(args) {
   if (args.length < 2) return
 
-  const command = args[1]
-  if (command === 'rotate') {
+  const options = args.slice(1)
+  if (options.some(o => o === 'rotate')) {
     if (mb.app.isReady()) {
       rotateUsers()
     } else {

